seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/utils/SchemaUtil.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.seatunnel.translation.spark.utils;
+
+import org.apache.seatunnel.api.table.catalog.CatalogTable;
+import org.apache.seatunnel.api.table.catalog.CatalogTableUtil;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.translation.spark.execution.ColumnWithIndex;
+import org.apache.seatunnel.translation.spark.execution.IndexQueue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class SchemaUtil {
+
+    public static ColumnWithIndex[] mergeSchema(List<CatalogTable> catalogTables) {
+        if (catalogTables.size() == 1) {
+            CatalogTable catalogTable = catalogTables.get(0);
+            return new ColumnWithIndex[] {
+                new ColumnWithIndex(
+                        IntStream.rangeClosed(
+                                        0, catalogTable.getSeaTunnelRowType().getTotalFields())
+                                .toArray(),
+                        catalogTable,
+                        null)
+            };
+        }
+        List<String> fieldNames = new ArrayList<>();
+        List<SeaTunnelDataType<?>> fieldTypes = new ArrayList<>();
+        List<ColumnWithIndex> columnWithIndexes = new ArrayList<>();
+        fieldTypes.sort(Comparator.comparingInt(o -> o.getSqlType().ordinal()));
```

Comment:
`fieldTypes` is empty. It's meaningless to sort it.

Reply:
It's not being used. I'll remove it. Thank you